### DISCUSSION TITLE
updating Many to Many relationships with connectOrCreate

### DISCUSSION THREAD

**Discussion by lebonMIA** [prisma/prisma#18219] [Q&A]
from the example schema in documentation :

```prisma
model Post {
  id         Int                 @id @default(autoincrement())
  title      String
  categories CategoriesOnPosts[]
}

model Category {
  id    Int                 @id @default(autoincrement())
  name  String
  posts CategoriesOnPosts[]
}

model CategoriesOnPosts {
  post       Post     @relation(fields: [postId], references: [id])
  postId     Int // relation scalar field (used in the `@relation` attribute above)
  category   Category @relation(fields: [categoryId], references: [id])
  categoryId Int // relation scalar field (used in the `@relation` attribute above)
  assignedAt DateTime @default(now())
  assignedBy String

  @@id([postId, categoryId])
}
```

I can insert a post with this code :

```ts
const postToCreate = {
    title: "TITLE",
    categories: [
        "CAT1",
        "CAT2",
    ]
}

await prisma.post.create({
    title: postToCreate.title,
    categories: {
        create: postToCreate.categories
        .map(h => ({
          category: {
            connectOrCreate: {
              where: {
                name: h,
              },
              create: {
                name: h,
              },
            },
          }
        }))
    });

```

Now I want to update the post.
With the following code :

```ts
const postToUpdate = {
    title: "TITLE",
    categories: [
        "CAT1",
        "CAT2",
        "CAT3",
    ]
}

await prisma.post.update({
    where: {id: XXX},
    data: {
        title: postToCreate.title,
        categories: {
            create: postToCreate.categories
            .map(h => ({
            category: {
                connectOrCreate: {
                where: {
                    name: h,
                },
                create: {
                    name: h,
                },
                },
            }
            }))
        }
    });
```

It (obviously) fails with the following error "Unique constraint failed on the fields: (`postId`,`categoryId`)"

What is the correct way to update the post ?

**Comment by nurul3101** [comment]
Hello @lebonMIA 👋 

Do you want to update a post and also simultaneously create categories? Or you just want to update a post?

**Reply by nurul3101** [reply]
@lebonMIA Did you had a chance to look at my response?
It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by lebonMIA** [reply]
Hi @nurul3101 sorry for the delay, I'm not sure to understand your question. I want to update a post's categories. Categories might exist or not, that's why I thought I could use _connectOrCreate_ to do that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
